cmd/cloud_upload.go

```diff
@@ -0,0 +1,189 @@
+	if err != nil {
+		return err
+	}
+
+	// Cloud config
+	cloudConfig, warn, err := cloudapi.GetConsolidatedConfig(
+		test.derivedConfig.Collectors["cloud"], c.gs.Env, "", arc.Options.Cloud, arc.Options.External)
+	if err != nil {
+		return err
+	}
+	if !cloudConfig.Token.Valid {
+		return errors.New( //nolint:golint
+			"not logged in, please login to the Grafana k6 Cloud " +
+				"using the `k6 cloud login` command",
+		)
+	}
+
+	// Display config warning if needed
+	if warn != "" {
+		modifyAndPrintBar(c.gs, progressBar, pb.WithConstProgress(0, "Warning: "+warn))
+	}
+
+	if cloudConfig.Token.Valid {
+		tmpCloudConfig["token"] = cloudConfig.Token
+	}
+	if cloudConfig.Name.Valid {
+		tmpCloudConfig["name"] = cloudConfig.Name
+	}
+	if cloudConfig.ProjectID.Valid {
+		tmpCloudConfig["projectID"] = cloudConfig.ProjectID
+	}
+
+	if arc.Options.External == nil {
+		arc.Options.External = make(map[string]json.RawMessage)
+	}
+
+	b, err := json.Marshal(tmpCloudConfig)
+	if err != nil {
+		return err
+	}
+
+	arc.Options.Cloud = b
+	arc.Options.External[cloudapi.LegacyCloudConfigKey] = b
+
+	name := cloudConfig.Name.String
+	if !cloudConfig.Name.Valid || cloudConfig.Name.String == "" {
+		name = filepath.Base(test.sourceRootPath)
+	}
+
+	logger := c.gs.Logger
+
+	// Start cloud test upload
+	modifyAndPrintBar(c.gs, progressBar, pb.WithConstProgress(0, "Validating script options"))
+	client := cloudapi.NewClient(
+		logger, cloudConfig.Token.String, cloudConfig.Host.String, consts.Version, cloudConfig.Timeout.TimeDuration())
+	if err = client.ValidateOptions(arc.Options); err != nil {
+		return err
+	}
+
+	modifyAndPrintBar(c.gs, progressBar, pb.WithConstProgress(0, "Uploading archive"))
+
+	var cloudTestRun *cloudapi.CreateTestRunResponse
+	cloudTestRun, err = client.UploadTestOnly(name, cloudConfig.ProjectID.Int64, arc)
+	if err != nil {
+		return err
+	}
+
+	refID := cloudTestRun.ReferenceID
+	if cloudTestRun.ConfigOverride != nil {
+		cloudConfig = cloudConfig.Apply(*cloudTestRun.ConfigOverride)
+	}
+
+	et, err := lib.NewExecutionTuple(test.derivedConfig.ExecutionSegment, test.derivedConfig.ExecutionSegmentSequence)
+	if err != nil {
+		return err
+	}
+	testURL := cloudapi.URLForResults(refID, cloudConfig)
+	executionPlan := test.derivedConfig.Scenarios.GetFullExecutionRequirements(et)
+	printExecutionDescription(
+		c.gs, "cloud", test.sourceRootPath, testURL, test.derivedConfig, et, executionPlan, nil,
+	)
+
+	if !c.gs.Flags.Quiet {
+		valueColor := getColor(c.gs.Flags.NoColor || !c.gs.Stdout.IsTTY, color.FgCyan)
+		printToStdout(c.gs, fmt.Sprintf(
+			"     test status: %s\n", valueColor.Sprint("Archived"),
+		))
+	} else {
+		logger.WithField("run_status", "Archived").Debug("Test uploaded")
+	}
+
+	return nil
+}
+
+func (c *cmdCloudUpload) flagSet() *pflag.FlagSet {
+	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
+	flags.SortFlags = false
+	flags.AddFlagSet(optionFlagSet())
+	flags.AddFlagSet(runtimeOptionFlagSet(false))
```

Comment:
We should revisit which of all of these still make sense for _just_ uploading the archive.